package ipc;

message IPCName
{
	required string module_name = 1;
	optional string host_name = 2;
	optional string suffix = 3;
}

message AddIPCObject
{
	required IPCName ipc_name = 1;
	required string ip = 2;
	required int32 port	= 3;
	required string access_id = 4;
}

message UpdateIPCObject
{
	required IPCName ipc_old_name	= 1;
	required IPCName ipc_new_name	= 2;
}

message ChangeIPCName
{
	required IPCName ipc_name = 1;
}

message IPCObjectList
{
	repeated AddIPCObject ipc_object = 1;
	optional string access_id = 2;
}

message ModuleName
{
	required IPCName ipc_name	= 1;
	required string ip			= 2;	//for ipc communication in local computer only
	required int32 port			= 3;	//for ipc communication in local computer only
	optional string access_id		= 4;	//for ipc communication in local computer only
	optional bool is_exist		= 5;	//for internal usage
	optional string conn_id		= 6;	//for internal usage
}

message ModuleState
{
	required bool exist = 1;
	required string rndVal = 2;
	optional string id = 3;
}

message RemoveIPCObject
{
	required string ipc_name = 1;
}

message IPCMessage
{
	required string message_name = 1;
	repeated IPCName ipc_path = 2;
	repeated IPCName ipc_sender = 3;
	optional bytes message = 4;
}

message Ping
{
}

option optimize_for = LITE_RUNTIME;